libs/react-components/tedi/src/components/tag/tag.tsx

```diff
@@ -0,0 +1,64 @@
+import cn from 'classnames';
+import { MouseEventHandler } from 'react';
+
+import { BreakpointSupport, useBreakpointProps } from '../../helpers';
+import ClosingButton from '../closing-button/closing-button';
+import { Icon } from '../icon/icon';
+import { Spinner } from '../spinner/spinner';
+import styles from './tag.module.scss';
+
+type TagColor = 'primary' | 'secondary' | 'danger';
+
+type TagBreakpointProps = {
+  /**
+   * Function to be called when the close button is clicked.
+   * If provided, a close button will be rendered inside the Tag.
+   */
+  onClose?: MouseEventHandler<HTMLButtonElement>;
+  /**
+   * Determines whether the Tag is in a loading state
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Determines Tag color
+   * When true, the Tag will be styled accordingly and an error icon will be displayed.
+   * @default 'primary'
+   */
+  color?: TagColor;
+};
+
+export interface TagProps extends BreakpointSupport<TagBreakpointProps> {
+  /**
+   * The content inside the Tag.
+   * Typically this will be text or any JSX elements to be displayed.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classes to apply custom styles to the Tag.
+   */
+  className?: string;
+}
+
+export const Tag = (props: TagProps): JSX.Element => {
+  const { getCurrentBreakpointProps } = useBreakpointProps();
+  const {
+    children,
+    className,
+    onClose,
+    isLoading = false,
+    color = 'primary',
+    ...rest
+  } = getCurrentBreakpointProps<TagProps>(props);
+
+  const tagBEM = cn(styles['tedi-tag'], color && styles[`tedi-tag--color-${color}`], className);
```

Comment:
Btw, why we use `tedi-tag` not just `tag`? Modular css already gives us special classnames. It just makes CSS longer and harder to read DOM

Reply:
This was to minimise the overriding styles of the Community Tag component, since we won't be removing it right away

Reply:
But there is no need really?

Reply:
Theoretically no, but TEDI team agreed to have tedi classes